contracts/RMRK/extension/tokenProperties/RMRKTokenPropertiesRepository.sol

```diff
@@ -0,0 +1,471 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+import "../../library/RMRKErrors.sol";
+import "./IRMRKTokenPropertiesRepository.sol";
+
+/**
+ * @title RMRKTokenPropertiesRepository
+ * @author RMRK team
+ * @notice Smart contract of the RMRK Token property repository module.
+ */
+contract RMRKTokenPropertiesRepository is IRMRKTokenPropertiesRepository {
+    mapping(address => mapping(uint256 => AccessType))
+        private _parameterAccessType;
+    mapping(address => mapping(uint256 => address))
+        private _parameterSpecificAddress;
+    mapping(address => IssuerSetting) private _issuerSettings;
+    mapping(address => mapping(address => bool)) private _collaborators;
+
+    // For keys, we use a mapping from strings to IDs.
+    // The purpose is to store unique string keys only once, since they are more expensive,
+    mapping(address => mapping(string => uint256)) private _keysToIds;
```

Comment:
There's no need to do it by collection. We can make it general and save some gas. If multiple collections use the same Key (which I suspect they will), only the first one creates the key, the rest are cheaper.

Reply:
Yeah, I considered this, but then started thinking about how this would behave through decades and decided to leave it 🤔 
It's an insanely huge number for the possible IDs, but still..
I can remove it if you think it's future-proof without doing it per collection 👌

Reply:
It's still 1.15e77 IDs :sweat_smile:  but seen that way, not strongly opposed to change it. We can also see what community thinks.

Reply:
This also means that we don't need to keep the `_totalProperties` value per-contract.. 🤔